lambda_function.py

```diff
@@ -1,57 +1,44 @@
 import boto3
 import socket
 import os
+import json
 
-# Define the clients at the top of your function
+# Initialize clients
 elbv2_client = boto3.client('elbv2')
 rds_client = boto3.client('rds')
 
-RDS_IDENTIFIER = os.environ['RDS_IDENTIFIER']  # RDS instance identifier
-TARGET_GROUP_ARN = os.environ['TARGET_GROUP_ARN']  # Target Group ARN
-
+# Load RDS details from environment variables
+RDS_DETAILS = json.loads(os.environ['RDS_DETAILS'])
 
 def lambda_handler(event, context):
-    # Retrieve the current IP address of the RDS instance
-    rds_instances = rds_client.describe_db_instances(
-        DBInstanceIdentifier=RDS_IDENTIFIER)
-    rds_endpoint = rds_instances['DBInstances'][0]['Endpoint']['Address']
-    ip_address = socket.gethostbyname(rds_endpoint)
-    rds_port = rds_instances['DBInstances'][0]['Endpoint']['Port']
-
-    # Retrieve the existing target of the target group
-    targets = elbv2_client.describe_target_health(
-        TargetGroupArn=TARGET_GROUP_ARN)
-
-    # Get the current IP address in the target group
-    if targets['TargetHealthDescriptions']:
-        current_ip = targets['TargetHealthDescriptions'][0]['Target']['Id']
-    else:
-        current_ip = None
-
-    # If the IP addresses don't match, update the target group
-    if current_ip and current_ip != ip_address:
-            # Deregister the current target
-            elbv2_client.deregister_targets(
-                TargetGroupArn=TARGET_GROUP_ARN,
-                Targets=[
-                    {
-                        'Id': current_ip
-                    },
-                ]
-            )
+    update_messages = []
+
+    for rds_identifier, details in RDS_DETAILS.items():
+        # Retrieve the current IP address of the RDS instance
+        rds_instances = rds_client.describe_db_instances(DBInstanceIdentifier=rds_identifier)
+        rds_endpoint = rds_instances['DBInstances'][0]['Endpoint']['Address']
+        ip_address = socket.gethostbyname(rds_endpoint)
+
+        # Retrieve the existing target of the target group
+        target_group_arn = details['target_group_arn']
+        targets = elbv2_client.describe_target_health(TargetGroupArn=target_group_arn)
+
+        # Check and update the target group
+        current_ip = targets['TargetHealthDescriptions'][0]['Target']['Id'] if targets['TargetHealthDescriptions'] else None
+        if current_ip != ip_address:
+            if current_ip:
+                # Deregister the current target
+                elbv2_client.deregister_targets(TargetGroupArn=target_group_arn, Targets=[{'Id': current_ip}])
```

Comment:
We should probably not let a failure in one DB result in all DB LBs not being configured correctly by this lambda. For example, if 5 RDS DBs were configured but one DB were destroyed rds_endpoint would fail with IndexError and stop the lambda in its tracks. 



Maybe some try catches or validation to ensure there are rds instances, alternatively the bulk of the logic could be moved into a colusre
```python
    success=True
    for rds_identifier, details in RDS_DETAILS.items():
        try:
            update_target_registration(rds_identifiers, details)
        except Exception as e:
            success = False
            message = f''Failed to update targets for {rds_identifier} with error: {e}"

```

Reply:
Thank you for the review @jubrad! Just made those changes, let me know what you think!

I refactored the function a little bit so that there is now a separate `update_target_registration` func, which has the logic for updating the target registration for a single RDS instance and then each iteration collects the info , eg. success or failure.